pkg/services/check/check.go

```diff
@@ -0,0 +1,115 @@
+package check
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/weaveworks/weave-gitops/pkg/flux"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+)
+
+var ErrFluxNotFound = errors.New("flux is not installed")
+var ErrKubernetesNotFound = errors.New("no kubernetes version found")
+
+const (
+	FluxCompatibleMessage    = "Current flux version is compatible"
+	FluxNotCompatibleMessage = "Current flux version is not compatible"
+)
+
+func Pre(ctx context.Context, kubeClient kube.Kube, fluxClient flux.Flux, expectedFluxVersion string) (string, error) {
+	output := ""
+
+	k8sOutput, err := validateKubernetes(fluxClient)
+	if err != nil {
+		return "", err
+	}
+
+	output += k8sOutput
+
+	currentFluxVersion, err := getCurrentFluxVersion(ctx, kubeClient)
+	if err != nil {
+		if errors.Is(err, ErrFluxNotFound) {
+			output += "✔ Flux is not installed"
+			return output, nil
+		}
+
+		return "", fmt.Errorf("failed getting current flux version: %w", err)
+	}
+
+	fluxOutput, err := validateFluxVersion(currentFluxVersion, expectedFluxVersion)
+	if err != nil {
+		return "", err
+	}
+
+	output += fluxOutput
+
+	return output, nil
+}
+
+func validateKubernetes(fluxClient flux.Flux) (string, error) {
+	output, err := fluxClient.PreCheck()
+	if err != nil {
+		return "", fmt.Errorf("failed running flux pre check %w", err)
+	}
+
+	lines := strings.Split(output, "\n")
+	for _, line := range lines {
+		if strings.Contains(line, "Kubernetes") {
+			return line + "\n", nil
+		}
+	}
+
+	return "", ErrKubernetesNotFound
+}
+
+func getCurrentFluxVersion(ctx context.Context, kubeClient kube.Kube) (string, error) {
+	namespacesList, err := kubeClient.GetNamespaces(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	for _, namespace := range namespacesList.Items {
```

Comment:
This is not is not in install path yet, maybe you got confused with PR #1002 . But I agree let's use that function instead.

Reply:
I'm having second thoughts.  I was adding the method `FetchNamespaceWithLabel` from PR #1002 . But then I realized I would need another method to get the label values. So I think the current method `GetNamespaces` allows me to do both at the same time. What do you thing about this?

Reply:
Sorry, my message wasn't clear.  Can we use the FetchNamespaceWithLabel() https://github.com/weaveworks/weave-gitops/blob/390fc7c3db7454d30be430459c8fec9997ceb24f/pkg/kube/kubehttp.go#L494 instead of duplicating the logic here?

Reply:
Looks like this new function just landed in main so when you rebase you will pick it up.

Reply:
@palemtnrider  That method only returns a namespace name. I need to get a specific label's value. There are two labels values I'm interested in: `app.kubernetes.io/part-of` and `app.kubernetes.io/version`. I am using the first label to find the right namespace. Once I have it. I can get the value for the second label. So then I can continue with the logic. Let me know if I'm missing something.

Reply:
Sorry, I should have picked up on that.  🤦